src/mito/src/engine.rs

```diff
@@ -381,11 +381,181 @@ fn validate_create_table_request(request: &CreateTableRequest) -> Result<()> {
     Ok(())
 }
 
+fn all_regions_open(table: TableRef, regions: &[RegionNumber]) -> TableResult<bool> {
+    Ok(regions
+        .iter()
+        .map(|r| table.contain_region(*r))
+        .collect::<TableResult<Vec<_>>>()?
+        .into_iter()
+        .all(|r| r))
+}
+
 impl<S: StorageEngine> MitoEngineInner<S> {
-    async fn open_table(
+    /// Returns Some(table) contains all specific regions
+    fn check_regions(
+        &self,
+        table: TableRef,
+        regions: &[RegionNumber],
+    ) -> TableResult<Option<TableRef>> {
+        if all_regions_open(table.clone(), regions)? {
+            // If all regions have been opened
+            Ok(Some(table))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Builds table from scratch.
+    /// Returns None if failed to recover manifest.
+    async fn build_table(
```

Comment:
build_tables sounds like creating a new table.
```suggestion
    async fn recover_table(
```